src/github.com/getlantern/flashlight/client/reverseproxy.go

```diff
@@ -40,23 +28,25 @@ func (client *Client) initReverseProxy(bal *balancer.Balancer, dumpHeaders bool)
 		DisableKeepAlives: true,
 	}
 
+	// Just choose a random dialer that also takes care of things like the
+	// authentication token.
+	dialer := client.getBalancer().RandomDialer()
```

Comment:
Hey @myleshorton, 

I noticed that `RandomDialer()` can return a `nil` function from `randomDialer` and that would trigger a random panic. That could not happen with `getReverseProxy()` because it waited for a valid dialer to arrive before returning.